tests/next_tests/integration_tests/feature_tests/iterator_tests/test_horizontal_indirection.py

```diff
@@ -70,7 +70,7 @@ def test_simple_indirection(program_processor):
 
     ref = np.zeros(shape, dtype=inp.dtype)
     for i in range(shape[0]):
-        ref[i] = inp.ndarray[i + 1 - 1] if cond[i] < 0.0 else inp.ndarray[i + 1 + 1]
+        ref[i] = inp.ndarray[i + 1 - 1] if cond.ndarray[i] < 0.0 else inp.ndarray[i + 1 + 1]
```

Comment:
```suggestion
        ref[i] = inp.asnumpy()[i + 1 - 1] if cond.asnumpy()[i] < 0.0 else inp.asnumpy()[i + 1 + 1]
```
to use the normal pattern (and do a first step in allocating for the proper device)